Sources/Controllers/MessagesViewController.swift

```diff
@@ -328,3 +366,35 @@ fileprivate extension MessagesViewController {
         return 0
     }
 }
+
+// MARK: - Menu Handling
+extension MessagesViewController {
+    /// Add observer for `UIMenuControllerWillShowMenu` notification
+    fileprivate func addMenuControllerObservers() {
+        NotificationCenter.default.addObserver(self, selector: #selector(MessagesViewController.menuControllerWillShow(aNotification:)), name: NSNotification.Name.UIMenuControllerWillShowMenu, object: nil)
+    }
+    
+    /// Remove observer for `UIMenuControllerWillShowMenu` notification
+    fileprivate func removeMenuControllerObservers() {
+        NotificationCenter.default.removeObserver(self, name: NSNotification.Name.UIMenuControllerWillShowMenu, object: nil)
+    }
+    
+    /// Show menuController and set target rect to selected bubble
+    @objc func menuControllerWillShow(aNotification:Notification) {
+        if let currentMenuController = aNotification.object as? UIMenuController,
+            let selectedIndexPath = selectedIndexPathForMenu {
+            NotificationCenter.default.removeObserver(self, name: NSNotification.Name.UIMenuControllerWillShowMenu, object: nil)
+            defer {
+                NotificationCenter.default.addObserver(self, selector: #selector(MessagesViewController.menuControllerWillShow(aNotification:)), name: NSNotification.Name.UIMenuControllerWillShowMenu, object: nil)
```

Comment:
Just a question, why do we keep adding ab observer here? It looks like we never remove it

Reply:
@nathantannar4 , emm, we need remove here first, because we call `currentMenuController.setMenuVisible(true, animated: true)` later, it would leads to endless loop if we not remove in advance.